codec/decode.go

```diff
@@ -1869,6 +1869,12 @@ type Decoder struct {
 	nsp *sync.Pool
 	err error
 
+	// Whether we're in a Decode or MustDecode call.
+	decoding bool
```

Comment:
Forgot to mention, it's safe to put these as members, since `Decoder` seems to not be goroutine-safe already

Reply:
Is the point that were no worse off than we were before?

Reply:
Yeap! If anyone was using the same Decoder from multiple goroutines, they would have surely hit the race detector.

Reply:
Can you put these comments into the code comments?

Reply:
Done, just added a note saying that Decoder is not goroutine-safe.